### ISSUE TITLE
Not compatible with React Hot Loader v4

### ISSUE THREAD

**Issue by chucksellick** [parcel-bundler/parcel#500]
React Hot Loader v4 removes the need for `react-hot-loader/patch`.

https://github.com/gaearon/react-hot-loader/tree/next#no-patch-required

When trying to use the new version in a Parcel project, I get this error in console and my app won't load:

![image](https://user-images.githubusercontent.com/1188786/34639214-2e4a19aa-f2d3-11e7-8def-7e42316cad96.png)

**Comment by theKashey** [comment]
Parcel does work with RHLv4. The only current issue is with Parcel's HRM -> module.hot.accepts works not as among the other bundlers.
RHL expects module or to be auto-accepted (webpack), or to contain `id` field(electron, webpack), with a file's name to be (re)required.
To use parcel with RHL just use RHL as v3, not v4. I mean - accept the changes from children in the "app.js", do not use `hot` function.

**Comment by jgkim** [comment]
Same here. Even if I don't use the `hot` function, I get the error `Cannot find module: 'react-hot-loader/patch'`.

**Comment by chucksellick** [comment]
Yes, something is trying to require `react-hot-loader/patch`, which doesn't exist anymore. This can't be a problem with how I'm using RHL. Something in Parcel must be doing this.

**Comment by theKashey** [comment]
Just double checked - everything works, node_modules/react-hot-loader/patch.js exists. This moment did not change from 3 to 4 version.

PS: "No-patch-required" is cos RHL adding it by itself via the babel plugin. It is still exists and needed :P

**Comment by jgkim** [comment]
The babel plugin adds `react-hot-loader/patch` automatically, but I cannot find any `require(‘react-hot-loader/patch’)` in the bundled file of the example of RHL. I guess that’s why it seems everything works.

**Comment by theKashey** [comment]
Double checked: 
- parcel 1.3.1 - not working
- parcel 1.4.1 - good!

**Comment by jgkim** [comment]
The same error is raised on parcel 1.4.1. The issue is not about using `hot`. The babel plugin puts some `require` statements and `node_modules/react-hot-loader/patch.js` exists, but the bundled file cannot find it. RHL v.3 works.

**Comment by theKashey** [comment]
This is not an issue RHL could solve, this is a something parcel should double check.
For 1.3.1 I've got an error about it could not require ./patch.prod, from the react-hot-loader/patch it could require. For 1.4.1 I've got an error - _could not find "react"_. On next run there was not issue.

Try just remove node_modules/(or even yarn.lock), and start from the scratch. It might be some caching issue.
